aspnet/security/2fa.rst

```diff
@@ -1,133 +1,109 @@
+
+.. Add Set up SMS for ASPSMS just after "Setup for Twilio"
 
 In this article:
 	- `Create a New ASP.NET 5 Project`_
-	- `Running the Application`_
-	- `Setup up SMS for Two-factor authentication`_
-	- `Enable Two-factor authentication`_
-	- `Login with Two-factor authentication`_
+	- `Setup up SMS for two-factor authentication with Twilio`_
+	- `Enable two-factor authentication`_
+	- `Login with two-factor authentication`_
 	- `Account lockout for protecting against brute force attacks`_
-	- `Next steps`_
-	- `Summary`_
+	- `Debugging  Twilio`_
 
 Create a New ASP.NET 5 Project
 ------------------------------
 
-To get started, open Visual Studio 2015. Next, create a New Project (from the Start Page, or via File - New - Project).  On the left part of the New Project window, make sure the Visual C# templates are open and "Web" is selected, as shown:
+Create a new ASP.NET 5 web app with individual user accounts.
 
-.. image:: 2fa/_static/new-project.png
+.. image:: accconfirm/_static/new-project.png
 
-Next you should see another dialog, the New ASP.NET Project window:
- 
-.. image:: 2fa/_static/select-project.png
-	
-Select the ASP.NET 5 Web site template from the set of ASP.NET 5 templates. Make sure you have Individual Authentication selected for this template. After selecting, click OK.
+Follow the instructions in :doc:`accconfirm` to set up and require SSL.
+
+Setup up SMS for two-factor authentication with Twilio
+-------------------------------------------------------
 
-At this point, the project is created. It may take a few moments to load, and you may notice Visual Studio's status bar indicates that Visual Studio id downloading some resources as part of this process.  Visual Studio ensures some required files are pulled into the project when a solution is opened (or a new project is created), and other files may be pulled in at compile time.
+ - Create a `Twilio <http://www.twilio.com/>`_ account.
+ - On the **Dashboard** tab of your Twilio account, note the **Account SID** and **Authentication token**. Note: Tap **Show API Credentials** to see the Authentication token.
+ - On the **Numbers** tab, note the Twilio phone number.
+ - Install the Twilio NuGet package. From the Package Manager Console (PMC),  enter the following the following command:
 
+ ``Install-Package Twilio``
+  - Add code in the *Services/MessageServices.cs* file to enable SMS.
 
-Running the Application
------------------------
+.. literalinclude:: 2fa/sample/WebSMS/src/WebSMS/Services/MessageServices.cs
+   :language: c#
+   :lines: 12-39
+   :dedent: 4 
 
-Run the application and after a quick build step, you should see it open in your web browser.
+.. comment note:: Twilio cannot target dnxcore50: You will get compilation errors if you build your project when dnxcore50 is included because Twilio does not have a package for dnxcore50. You can remove dnxcore50 from the *project.json* file or you can call the Twilio REST API to send SMS messages.
```

Comment:
You can point to this blog to get Twilio running on .NET Core https://www.twilio.com/blog/2015/08/getting-started-with-asp-net-5-and-visual-studio-code-on-a-mac.html